e2-studio-star-rx72n-firmware/libs/rx_hal/src/rx_mpc.c

```
@@ -1089,3 +1089,62 @@ rx_err_t rx_mpc_set_usb_vbus(const rx_port_pin_t pin)
+ * @note Used by HC-SR04 ultrasonic sensors for echo pulse measurement
+ *
+ * @see rx_mpc_set_adc() Same pattern: writes ASEL bit (0x80) directly
+ * @see rx_mpc.h Full API documentation with usage examples
+ * @see rx_hcsr04_icu_configure() Configure ICU after calling this function
+ *
+ * @since Version 1.2.0 (Issue #296 - IRQ-based HC-SR04 measurement)
+ */
+rx_err_t rx_mpc_set_irq(const rx_port_pin_t pin)
+{
+  /* Extract port and pin number from type-safe enum */
+  const uint8_t port    = rx_port_from_pin(pin);
```

Clang (14.0.6)

[warning] variable 'port' is not initialized

(cppcoreguidelines-init-variables)